File: 0682/sesión 15  - Master Detail/sin Master Detail/SoftInv/SoftInvTest/DBManagerTest.cs
using System;
using System.Data.Common;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoftInv.Db;

namespace SoftInvTest
{
    [TestClass]
    public class DBManagerTest
    {
        [TestMethod]
        public void TestGetInstance()
        {
            Console.WriteLine("GetInstance");
            DBManager dBManager1 = DBManager.Instance;
            DBManager dBManager2 = DBManager.Instance;
            Assert.IsNotNull(dBManager1);
            Assert.IsNotNull(dBManager2);
            Assert.AreEqual(dBManager1, dBManager2);
        }

        [TestMethod]
        public void TestGetConnection()
        {
            Console.WriteLine("GetConnection");
            DbConnection conexion = DBManager.Instance.Connection;
            Assert.IsNotNull(conexion);
            conexion.Open();
        }
    }
}

File: 0682/sesión 22 - ex1_2025_2_P3/Pregunta 03 - Estudiantes/Pregunta3_ProyectoBase/TransitSoftDBManager/src/test/java/pe/pucp/transitsoft/db/DBManagerTest.java
package pe.pucp.transitsoft.db;

import java.sql.Connection;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class DBManagerTest {
    
    @Test
    public void testGetInstance() {
        System.out.println("getInstance");                
        DBManager dBManager = DBManager.getInstance();
        assertNotNull(dBManager);
    }

    @Test
    public void testGetConnection() {
        System.out.println("getConnection");                
        DBManager dBManager = DBManager.getInstance();
        Connection conexion = dBManager.getConnection();
        assertNotNull(conexion);
    }
}
